parser_library/src/processing/statement_providers/members_statement_provider.cpp

```diff
@@ -128,8 +140,11 @@ context::shared_stmt_ptr members_statement_provider::preprocess_deferred(
 
     const auto& cache_item = cache.get(status.first.form);
 
-    for (const diagnostic_op& diag : cache_item->diags)
-        diagnoser.add_diagnostic(diag);
+    if (processor.kind != processing_kind::LOOKAHEAD)
+    {
+        for (const diagnostic_op& diag : cache_item->diags)
```

Comment:
I think not just yet. Because from what I observed, there are diagnostics that are printed once (e.g. during macro definition pass) and then there are diagnostics produced for each statement execution.
The current implementation tries to follow the behavior as closely as possible, but I think that we will need further refinement of how we store/replay diagnostics.